core/src/services/redis/backend.rs

```diff
@@ -299,26 +290,39 @@ impl Debug for Adapter {
 }
 
 impl Adapter {
-    async fn conn(&self) -> Result<RedisConnection> {
-        Ok(self
+    async fn conn(&self) -> Result<bb8::PooledConnection<'_, RedisConnectionManager>> {
```

Comment:
Hi, thanks a lot for this PR first. Have you communicated with the upstream first? I think it's a great feature that upstream might want to merge.